docs: split out "landing your work" section

brian-smith-tcril opened the pull request:
https://docsopenedxorg--472.org.readthedocs.build/en/472/developers/references/developer_guide/process/landing-your-work.html

sarina commented on source/developers/references/developer_guide/process/landing-your-work.rst:
```diff
@@ -0,0 +1,135 @@
+
+All code in the pull request must be compatible with the Open edX project's AGPL
+license.  This means that the author of the pull request must sign a
+`contributor's agreement`_, and all libraries included or
```
```suggestion
`contributor's agreement`_ (or be covered by their employer's agreement), and all libraries included or
```
Adding this because I know in the past we've had some slightly confused engineers at other companies submit a CLA when they don't need to.

sarina commented on source/developers/references/developer_guide/process/landing-your-work.rst:
```diff
@@ -0,0 +1,135 @@
+-------
+
+All of the tests must pass. You'll see the checks that are being run on the
+bottom of your PR; they must be green.
```
Possibly link to https://docsopenedxorg--472.org.readthedocs.build/en/472/developers/references/running_pr_tests.html here?

sarina commented on source/developers/references/developer_guide/process/landing-your-work.rst:
```diff
@@ -0,0 +1,135 @@
+`Paragon pattern library`_.
+
+-----------
+Code review
```
```suggestion
Code Review
```

sarina reviewed (approved):
3 small comments, looks great!

brian-smith-tcril commented:
@sarina I addressed all comments here https://github.com/openedx/docs.openedx.org/compare/2dd2b05e5785e3eb37b5337e4c4866f079e3fb3c..0f5a938ec1042700dd4c011905d7bbf559e01313

sarina commented:
👍🏻
